test/integration_test.rb

```diff
@@ -39,42 +39,6 @@ class IntegrationTest < Minitest::Test
     assert times[2] - times[1] > 1
   end
 
-  it 'streams async' do
-    next unless server.thin?
-
-    Timeout.timeout(3) do
-      chunks = []
-      server.get_stream '/async' do |chunk|
-        next if chunk.empty?
-        chunks << chunk
-        case chunk
-        when "hi!"   then server.get "/send?msg=hello"
-        when "hello" then server.get "/send?close=1"
-        end
-      end
-
-      assert_equal ['hi!', 'hello'], chunks
-    end
-  end
-
-  it 'streams async from subclass' do
-    next unless server.thin?
-
-    Timeout.timeout(3) do
-      chunks = []
-      server.get_stream '/subclass/async' do |chunk|
-        next if chunk.empty?
-        chunks << chunk
-        case chunk
-        when "hi!"   then server.get "/subclass/send?msg=hello"
-        when "hello" then server.get "/subclass/send?close=1"
-        end
-      end
-
-      assert_equal ['hi!', 'hello'], chunks
-    end
-  end
-
```

Comment:
Instead of removing these tests, I would like you to enable rainbows and check that works well.

You may not need to enable rainbows for all tests. It would be nice if rainbows are enabled only for these tests.

Reply:
I fixed test.
https://github.com/sinatra/sinatra/pull/1627/files#diff-09a9f659af53d3710c6cf95a5878bc5aR76

Reply:
I think this replacement goes well but it is not executed probably because rainbows is not installed by our Gemfile.
Also, not only that, we need to make the sinatra application for testing the streaming feature use rainbows.

Originally, our test had been using thin, but the tests haven't been executed since [I dropped the support](https://github.com/sinatra/sinatra/commit/328f66933ddc1cfab438835634c1aff9c2bec92d).
Could you follow up on this?

Reply:
I added [`server.unshift 'rainbows'` here](https://github.com/sinatra/sinatra/blob/607346295924c75a86f12ecbad9430cfc430c032/lib/sinatra/base.rb#L1849) to try it out but test failed because same reason as https://github.com/sinatra/sinatra/pull/1627#discussion_r461622715.

> sinatra/lib/sinatra/base.rb:1754:in `detect_rack_handler': Server handler (rainbows) not found. (RuntimeError)

`thin` could be used, but `rainbows` could not be used.

Could you tell me how to deal with this?

Reply:
IIUC, to deal with this, there is only way that defining and adding `Rack::Handler::Rainbows` for our test (please take care that it must not be loaded outside our test).

To be honest, there is not way for realizing that in a beautiful way, but I would prefer prioritizing the actual benefit over the apparently beauty.

What do you think? I would also love to hear your opinion on the issue.

Reply:
Sorry for the delay.

I tried to run the test with Rainbows( 994c9f5 ), but there is a problem that I couldn't solve.

It has to load `Rack::Handler::Rainbows` such as 3eb1003 .

Could you tell me how to deal with this?

Reply:
I'll try to create an independent test by not including rainbows in [`Sinatra server`](https://github.com/sinatra/sinatra/blob/3eb10038fd8caa24b1e754dc1e65164cba591f2f/lib/sinatra/base.rb#L1852)
I'll try to fix it.

Reply:
@namusyaka  I moved `'streams async'` and `'streams async from subclass'` test cases from here to `test/integration_async_test.rb`.
96d1f0b